src/FirebirdSql.Data.FirebirdClient/FirebirdClient/FbDataReader.cs

```diff
@@ -596,12 +596,12 @@ public override object GetValue(int i)
 				}
 				if (nullableUnderlying == typeof(bool))
 				{
-					return GetFieldValue<bool>(i);
+					return _row[i].GetBoolean();
```

Comment:
This is now not having checks `GetFieldValue` is doing. Also it hinders maintainability - if ever there's added logic for reading `bool` (or more checks) it would need to be copied here.

Reply:
My opinion is, that it may be better done in DbValue.
But in GetValue() it isn't performance relevant.